src/qforte/expansions/manifolds.py

```diff
@@ -0,0 +1,70 @@
+"""
+Quality-of-life shortcuts to build manifolds for excited state methods
+"""
+import copy
+from itertools import combinations
+
+def ipea_manifold(ref, n_ann, n_cre, sz_delta):
+    """Compute all excitations where n occupied orbitals are annihilated
+      and n_cre virtual electrons are created
+      sz_delta is a list of allowed spin changes.
+      (e.g. sz_delta = -1 would allow i->ab') 
+      """
+    M = len(ref)
+    occs = [i for i in range(M) if ref[i] == 1]
+    noccs = [i for i in range(M) if ref[i] == 0]
+    annihilate = combinations(occs, n_ann)
+    create = combinations(noccs, n_cre)
+    dets = []
+
+    for ann in annihilate:
+        for cre in copy.copy(create):
```

Comment:
You need the copy because combinations() gives an iterator that is "consumed" when you iterate through it- I can't iterate through the same instance more than once.